instrumentation/testing/coverage.go

```diff
@@ -0,0 +1,90 @@
+package testing
+
+import (
+	"fmt"
+	"github.com/google/uuid"
+	"go/build"
+	"os"
+	"sync/atomic"
+	"testing"
+	_ "unsafe"
+)
+
+type (
+	coverage struct {
+		Type    string         `json:"type" msgpack:"type"`
+		Version string         `json:"version" msgpack:"version"`
+		Uuid    string         `json:"uuid" msgpack:"uuid"`
+		Files   []fileCoverage `json:"files" msgpack:"files"`
+	}
+	fileCoverage struct {
+		Filename   string  `json:"filename" msgpack:"filename"`
+		Boundaries [][]int `json:"boundaries" msgpack:"boundaries"`
+	}
+)
+
+//go:linkname cover testing.cover
+var (
+	cover    testing.Cover
+	counters map[string][]uint32
+)
+
+// Clean the counters for a new coverage session
+func (test *Test) startCoverage() {
+	if counters == nil {
+		counters = map[string][]uint32{}
+	}
+	for name, counts := range cover.Counters {
+		counters[name] = make([]uint32, len(counts))
+		for i := range counts {
+			counters[name][i] = counts[i]
+			counts[i] = 0
+		}
+	}
+}
+
+// Get the counters values and extract the coverage info
+func (test *Test) endCoverage() *coverage {
+	gopath := os.Getenv("GOPATH")
+	if gopath == "" {
+		gopath = build.Default.GOPATH
+	}
+
+	fileMap := map[string][][]int{}
+	var active, total int64
+	var count uint32
+	for name, counts := range cover.Counters {
+		blocks := cover.Blocks[name]
+		for i := range counts {
+			stmts := int64(blocks[i].Stmts)
+			total += stmts
+			count = atomic.LoadUint32(&counts[i])
+			atomic.StoreUint32(&counts[i], counters[name][i]+count)
+			if count > 0 {
+				active += stmts
+				file := fmt.Sprintf("%v/src/%v", gopath, name)
```

Comment:
Is this always true? With go modules you can have your code outside of `GOPATH` right?

Reply:
Yes, you’re right, i have to implement the same algorithm used by go tool to find the files as you can see here:  https://github.com/golang/go/blob/master/src/cmd/cover/func.go#L178

findpkg and findfile. But they use internally go list to accomplish that...

Reply:
Done, the algorithm is implemented.

Reply:
https://shared.scope.dev/dashboard/8cee2622-8772-4338-8d72-63653e0e0239/66857b5a-cd46-4b9a-a6b3-438a51b84b4b/default/explore/inspect/8b105c3ee56fc7881e63f009c44b5fd59f32aca6/00000000-0000-0000-5a78-3c86d4e87f89/path